src/main/java/com/gachigang/ontherun/service/ProductService.java

```diff
@@ -46,9 +46,16 @@ public Product createProduct(Product product){
     }
 
     @Transactional
-    public Product hideProduct(Long id){
-        Product product = productRepository.findById(id).orElseThrow(()-> new EntityNotFoundException("Product not found with id: " + id));
-        product.setActive(false);
+    public Product updateProduct(Product product){
+        if (!productRepository.existsProductById(product.getId())){
```

Comment:
I don`t see any reasons to rewrite this part of code ... I guess it is better to have optional here